### ISSUE TITLE
Is there a sniff to disallow "variable variable"?

### ISSUE THREAD

**Issue by ghnp5** [slevomat/coding-standard#1315]
Looking to disallow things like:

    $$variable

With two dollar signs.

Thank you!

**Comment by janedbal** [comment]
This should be super simple with custom phpstan rule:
```php

use PhpParser\Node\Expr\Variable;
use PhpParser\Node;
use PHPStan\Analyser\Scope;
use PHPStan\Rules\Rule;

/**
 * @implements Rule<Variable>
 */
class CamelCaseFieldNamingRule implements Rule
{

    public function getNodeType(): string
    {
        return Variable::class;
    }

    /**
     * @param Variable $node
     * @return string[]
     */
    public function processNode(Node $node, Scope $scope): array
    {
        if (!is_string($node->name)) {
            return ['Expression for variable name is not allowed!'];
        }

        return [];
    }

}
```

**Comment by ghnp5** [comment]
Thank you very much.
Haven't got the chance to implement phpstan in my project yet!
But I'm taking the note for when I do :)

Do you think this would be hard to implement as a PHPCS rule?

**Comment by kukulich** [comment]
> Do you think this would be hard to implement as a PHPCS rule?

No, it should be really simple sniff :)

**Comment by ghnp5** [comment]
I leave the feature request here, so! Thank you!

**Comment by mabar** [comment]
Btw, it's already implemented by phpstan strict rules

https://phpstan.org/r/2966deda-b4e9-4d76-ada9-2c9b6467c857

**Comment by kukulich** [comment]
Implemented in https://github.com/slevomat/coding-standard/commit/43904bdc6c8f6d424cc75a582f47a52d68643081
